### DISCUSSION TITLE
How do I clear "React Hook useEffect has a missing dependency?"

### DISCUSSION THREAD

**Discussion by paulhoule** [react-hook-form/react-hook-form#11611] [Q&A] [answered]
So I am getting a setValue object from calling useForm,  a few line down I have a function like

```
   useEffect(() => {
      setValue("x", x)
   }, [x[)
```

and get the warning message

```
React Hook useEffect has a missing dependency: 'setValue'. Either include it or remove the dependency array
```

What's the best way to deal with this?

**Comment by Moshyfawn** [comment, marked as answer]
Add `setValue` to the dependency array. The method should be stable and not cause unnecessary re-rendering.

**Reply by dlibis** [reply]
@Moshyfawn  are all methods are stable enough to add to the deps? (methods from useFieldArray - remove,append,update; getValues and so on...)

**Reply by jgramling01** [reply]
> are all methods are stable enough to add to the deps? (methods from useFieldArray - remove,append,update; getValues and so on...)

@Moshyfawn I'd also like to know this. Is there any documentation that states this?
